sdk/core/core-http/src/serviceClient.ts

```diff
@@ -707,6 +708,10 @@ export function createPipelineFromOptions(
 
   requestPolicyFactories.push(logPolicy(loggingOptions));
 
+  if (isNode && !pipelineOptions.decompressResponse) {
+    requestPolicyFactories.push(disableResponseDecompressionPolicy());
```

Comment:
Do we need to add this to the pipelines for keyvault/app-configuration (basically any of the track2 libraries that might be using their own pipeline instead of the default?)

Reply:
I don't know whether any of the libraries would have this issue.  Blob has the issue because the service allows setting the content encoding of blobs. Other service might not even have the header when we retrieve things from them.  I plan to have the storage fix in a separate PR.  If we could think of any service that suffers the same issue I could include fixes there too.

Reply:
So the resource on the service side would need to have `Content-Encoding` set to either gzip or deflate, then node-fetch would decompress by default.